lib/ddtrace/ext/transport.rb

```diff
@@ -3,6 +3,7 @@ module Datadog
   module Ext
     module Transport
       module HTTP
+        ADAPTER = :net_http
```

Comment:
Because I had to add a few new constants here, I ended up performing a small refactor around transport constants, as you'll see in the files to follow.

Reply:
I think it makes sense to have it as only `:http`, but we can't really change it at this time because `:net_http` is publicly documented: https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#using-the-nethttp-adapter

We could have alternative http implementations (we had an Ethon implementation, for example), thus they might have more implementation specific names, but none today.

I'll document the desire to change `:net_http` to just `:net` here.

Reply:
Right! I think I was leaving out a bit too much detail on my description; let me share a bit more of my thinking, as I think it factors in some of my other feedback that you replied to as well.

I think the transport options proc really needs to go. Because it's effectively a bit of code that can do anything at all, it makes it really hard for us to know what's up and evolve the configuration. I think the direction we want to evolve in is one where those options are provided in the settings object, and the `AgentSettingsResolver` is able to convert everything into the resulting `AgentSettings` object.

Of course, until we actually do such a step, we'll need to still support the current public APIs and ways that people use to configure the gem.

BUT, and here is where some of my suggestions come in, nothing really forces us to use the "deprecated way" of doing things as the primary internal representation for them. E.g.

* We may support people using `t.adapter :net_http` in a proc, but just alias to `:http` 
* We may change some of the APIs to use `AgentSettings` objects directly, as long as they still support the "old" way of doing it as well

The advantage of doing this is that we push the "deprecated way" further and further to what we do and how we think about it, rather than considering it as set in stone and building around it. Conceptually think of it as reimplementing stuff from scratch to use the new way only AND THEN adding support for the deprecated way on the side VS leaving the deprecated way as the main way AND THEN adding the new way on the side :)

Now, of course we all have dreams and aspirations for the codebase, so feel free to draw a line and declare "this is how far I want to get in **_this_** PR, and we'll see about the rest in the future". I'm just sharing this to contextualize how I'm thinking about these things.

Reply:
> We could have alternative http implementations (we had an Ethon implementation, for example), thus they might have more implementation specific names, but none today.

On this specific bit, I would represent ethon to customers as an extra option to the http adapter, not as an http adapter as itself. E.g. I envision that customers ask for http, and we pick what's the best http option, and that at some point may be ethon.

Otherwise we create this situation where customers need to both upgrade AND THEN review their configuration regularly in order to take advantage of improvements; whereas I think a model where they upgrade and automatically get the latest and greatest ™️   provides a better experience.

Reply:
> On this specific bit, I would represent ethon to customers as an extra option to the http adapter, not as an http adapter as itself. E.g. I envision that customers ask for http, and we pick what's the best http option, and that at some point may be ethon.
>
> Otherwise we create this situation where customers need to both upgrade AND THEN review their configuration regularly in order to take advantage of improvements; whereas I think a model where they upgrade and automatically get the latest and greatest ™️ provides a better experience.

Totally agree, this is what I was thinking: using simply `:http` instead of `:net_http`, regardless of the underlying implementation.

>   * We may support people using t.adapter :net_http in a proc, but just alias to :http

That seems reasonable. I'm not sure it has to be done here. I'm thinking we can rip off the band-aid and rename `:net_http` to `:http` in 1.0 and that's it.

>   * We may change some of the APIs to use AgentSettings objects directly, as long as they still support the "old" way of doing it as well

I did some of this refactor, take a look!